app/views/account/memberships/show.html.erb

```diff
@@ -46,7 +46,9 @@
           <% else %>
             <%= link_to t('.buttons.promote'), [:promote, :account, @membership], method: :post, data: { confirm: t('global.confirm_message') }, class: first_button_primary if can? :promote, @membership %>
           <% end %>
-          <%= button_to t(".buttons.#{membership_destroy_locale_key(@membership)}"), [:account, @membership], method: :delete, data: { confirm: t(".buttons.confirmations.#{membership_destroy_locale_key(@membership)}", model_locales(@membership)) }, class: first_button_primary if can? :destroy, @membership %>
+          <% if (can? :destroy, @membership) && (!@membership.platform_agent?) %>
```

Comment:
We need this check to be part of `can?` somehow, otherwise people could destroy these via API or some other channel. Going to summon @adampal to help on this one when he has time.